processing/src/main/java/io/druid/segment/incremental/IncrementalIndex.java

```diff
@@ -353,7 +362,8 @@ protected abstract Integer addToFacts(
       InputRow row,
       AtomicInteger numEntries,
       TimeAndDims key,
-      ThreadLocal<InputRow> in
+      ThreadLocal<InputRow> rowContainer,
```

Comment:
this is needed so that OnHeapIncrementalIndex can set the next row. It wouldn't be possible to do the "set" on Supplier .